Firestore/Source/API/FIRBsonBinaryData.mm

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/Source/Public/FirebaseFirestore/FIRBsonBinaryData.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation FIRBsonBinaryData
+
+- (instancetype)initWithSubtype:(uint8_t)subtype data:(NSData *)data {
+  self = [super init];
+  if (self) {
+    _subtype = subtype;
+    _data = data;
```

Comment:
```objc
@property(atomic, copy, readonly) NSData *data;
```

When initializing a property with the copy attribute, the property setter that does the copy doesn't run when accessing the ivar with the _ prefix. So the copy needs to be manual here:

```suggestion
    _data = [data copy];
```

Good article: https://furbo.org/2012/05/04/arc-and-copy/

Since this and the other types are immutable and don't re-set the properties after initialization, the `copy` isn't doing anything. I'm okay with removing it or leaving it as-is as documentation of the intended ownership strategy.

Reply:
oh! TIL! keeping the `copy`. Done